presto-raptor/src/main/java/com/facebook/presto/raptor/util/DatabaseUtil.java

```diff
@@ -134,6 +142,31 @@ public static boolean isSyntaxOrAccessError(Exception e)
         return sqlCodeStartsWith(e, "42");
     }
 
+    public static boolean isTransactionCacheFullError(Exception e)
+    {
+        return mySqlErrorCodeMatches(e, ER_TRANS_CACHE_FULL);
+    }
+
+    /**
+     * Check if an exception is caused by a MySql exception of certain error code
+     */
+    private static boolean mySqlErrorCodeMatches(Exception e, int errorCode)
+    {
+        return Throwables.getCausalChain(e).stream()
+                .filter(SQLException.class::isInstance)
+                .filter(t -> ((SQLException) t).getErrorCode() == errorCode)
```

Comment:
Add a `.map(SQLException.class::cast)` before this, then you can remove the cast